src/paymentservice/src/main/java/ca/bc/gov/open/pssg/rsbc/dps/paymentservice/PaymentClient.java

```diff
@@ -0,0 +1,42 @@
+package ca.bc.gov.open.pssg.rsbc.dps.paymentservice;
+
+import java.net.URL;
+
+import ca.bc.gov.open.pssg.rsbc.dps.paymentservice.PaymentServiceConstants.BamboraTransType;
+import ca.bc.gov.open.pssg.rsbc.dps.paymentservice.exception.PaymentServiceException;
+
+/**
+ * 
+ * Payment Client interface
+ * 
+ * @author smillar
+ *
+ */
+public interface PaymentClient {
+
+	/**
+	 * 
+	 * Single Payment
+	 * 
+	 * @param correlationId
+	 * @param transType
+	 * @param invoiceNumber
+	 * @param totalItemsAmount
+	 * @param totalGST
+	 * @param totalPST
+	 * @param approvedPage
+	 * @param declinedPage
+	 * @param errorPage
+	 * @param ref1
+	 * @param ref2
+	 * @param ref3
+	 * @return
+	 * @throws PaymentServiceException
+	 */
+	public URL calculateSinglePaymentURL(String correlationId, BamboraTransType transType, String invoiceNumber,
```

Comment:
it's a lot of input params, I think sonarqube will give us a code smell here.

Reply:
100% will

Reply:
could we bundle the params in a json object and call a validate function to ensure that they're all there? how could we resolve this?

Reply:
I can try to break this up into a few methods but the problem is the call to Bambora takes them all. I will set my self a task to deal with this one later, ok? Might be able to do as Adam suggests and pass in a single param containing all info.